src/lib.rs

```diff
@@ -236,61 +236,64 @@ pub async fn bootstap(mut shutdown: broadcast::Receiver<()>, config: Config) ->
             );
 
         let app = Router::new()
-            .route("/health", get(handlers::health::handler))
-            .nest("/tenants", tenancy_routes)
+            .route("/health", get(handlers::health::handler).layer(
+                axum::middleware::from_fn_with_state(state_arc.clone(), rate_limit_middleware),
+            ))
+            .nest("/tenants", tenancy_routes.layer(
+                axum::middleware::from_fn_with_state(state_arc.clone(), rate_limit_middleware),
+            ))
             .route(
                 "/:tenant_id/clients",
-                post(handlers::register_client::handler),
+                post(handlers::register_client::handler).layer(
+                    axum::middleware::from_fn_with_state(state_arc.clone(), rate_limit_middleware),
+                ),
             )
             .route(
                 "/:tenant_id/clients/:id",
-                delete(handlers::delete_client::handler),
+                delete(handlers::delete_client::handler).layer(
+                    axum::middleware::from_fn_with_state(state_arc.clone(), rate_limit_middleware),
+                ),
             )
+            // Rate limiting middleware is not applying to push_handler because it is used by the relay
             .route(
                 "/:tenant_id/clients/:id",
                 post(handlers::push_message::handler),
             )
             .layer(global_middleware);
-
-        let app = if let Some(geoblock) = state_arc.geoblock.clone() {
```

Comment:
Removing this part as we are applying the geoblock and state later and not depending on the multitenant feature. This is a fix to double applying the geoblock and state.